addons/pos_loyalty/static/src/overrides/components/payment_screen/payment_screen.js

```diff
@@ -80,7 +80,10 @@ patch(PaymentScreen.prototype, {
                 points: pe.points - order._getPointsCorrection(ProgramModel.get(pe.program_id)),
             });
             const program = ProgramModel.get(pe.program_id);
-            if (program.is_nominative && partner) {
+            if (
```

Comment:
Override the `preSyncAllOrders` and verify if there is a partner before syncing the order.

`partner_id` can be undefined

Reply:
The `preSyncAllOrders` method is already overridden in the `pos_loyalty` module. When I tried to print the order, I did get the `partner_id` in the order (if we select a customer while ordering in POS).

If we make a POS order without a customer, then the `partner_id` is undefined. As a result, the condition `(program.is_nominative || program.program_type == "next_order_coupons") && partner)` is false because `partner` is undefined. Therefore, the coupon will not have any partner set to it.

(Ps. - I am not aware of the POS flows, please correct me if I have misunderstood anything)

Reply:
Only loyalty.card (coupon) objects that are created for "ewallet" and "loyalty" programs are required to have partner.

Reply:
Yes, but the task was to set a partner for next_order_coupons.